modules/core/src/main/java/org/apache/ignite/configuration/CacheConfiguration.java

```diff
@@ -185,6 +185,9 @@
     /** Default query parallelism. */
     public static final int DFLT_QUERY_PARALLELISM = 1;
 
+    /** Default query timeout. */
+    public static final long DFLT_QRY_TIMEOUT = 5000;
```

Comment:
I believe we should discuss configuration defaults changes on dev-list at first and preserve default behavior if possible.

Let's make it zero by default and treat zeroed value as Infinite Timeout to preserve previous behavior.
May be we can bound this later after discussion.